nntrainer/tensor/manager.h

```diff
@@ -372,6 +413,9 @@ class Manager {
   std::vector<std::unique_ptr<Var_Grad>>
     tensors_v2; /**< extra tensors required by the layers */
 
+  std::array<std::vector<std::unique_ptr<Var_Grad>>, NUM_TENSOR_GROUP_TYPE>
```

Comment:
how will you store std::unique_ptr<Weight> in this array?

Reply:
Weight is Var_Grad.
When accessing weight, we can static_cast before return.